File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.go
func tribonacci(n int) int {
    if n < 1 {
        return 0
    }

    if n < 3 {
        return 1 
    }
    
    dp := make([]int, n+1)
    dp[1] = 1
    dp[2] = 1
    for i := 3; i < n+1; i++ {
        dp[i] = dp[i-1] + dp[i-2] + dp[i-3]
    }

    return dp[n]
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution(object):
    def tribonacci(self, n):
        """
        :type n: int
        :rtype: int
        """
        dp = [0] * (n + 1)

        for i in range(1, n + 1):
            if i < 3:
                dp[i] = 1
            else:
                dp[i] = sum(dp[i-3:i])
            
        return dp[-1]
        